netket/_driver.py

```diff
@@ -0,0 +1,263 @@
+import sys
```

Comment:
can we call this _variational_driver.py or something to avoid confusion though?

Reply:
We probably should agree on a permanent place to put the new driver (at least until NetKet 3.0).
If we just follow the common scheme, we could just put it under `netket.variational.VmcDriver`.

Reply:
I don't recall exactly the outcome of our previous discussions, but I thought there was agreement about putting all the highest-level applications in the main namespace? something like `netket.Vmc` ? In any case, my comment was more about naming the python file something different than just `netket/_driver.py` that could cause ambiguity with other drivers

Reply:
Fair enough, I can rename it (after all, the name of that file is an implementation detail and can be changed without breaking the interface anyway) and import the driver directly in the `netket` module.

Reply:
Done. Let me know what you think of the interface now.
The new driver is `netket.Vmc`.

Also, I have changed the top-level `estimate_expectations` function, it can now be used like this:
https://github.com/netket/netket/blob/59e75baeb8476b875e2da15ca977351e5196b966/Test/GroundState/test_vmc.py#L73-L78
(It has an optional `compute_gradients` argument. Only downside is that it will compute the gradients for all observables. However, for more control it is always possible to use the lower-level functions like the driver does.)